q2_assembly/quast/tests/test_quast.py

```diff
@@ -705,6 +732,66 @@ def test_evaluate_contigs_action_paired_end_no_icarus(self, p1, p2, p3, p4):
         }
         p2.assert_called_once_with(ANY, self._tmp, context=exp_context)
 
+    @patch("pandas.read_csv")
+    @patch("q2_assembly.quast._parse_columns")
+    def test_create_tabular_results(self, p1, p2):
+        temp_dir = MockTempDir()
+        report_path = os.path.join(
+            temp_dir.name, "combined_reference", "transposed_report.tsv"
+        )
+        mock_df = pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
+        p2.return_value = mock_df
+
+        _ = _create_tabular_results(temp_dir.name)
+
+        p2.assert_called_once_with(report_path, sep="\t", header=0)
+        p1.assert_called_once_with(mock_df)
+
+    def test_evaluate_contigs_pipeline(self):
+        # this is used for mocking
+        def _copy_tsv_file(src, dst_dir):
+            if not os.path.exists(dst_dir):
+                os.makedirs(dst_dir)
+
+            destination_path = os.path.join(dst_dir, "quast_results.tsv")
+            shutil.copy2(src, destination_path)
+
+        input_files = self.get_data_path("contigs")
+        _input = ContigSequencesDirFmt(input_files, mode="r")
+        contigs = Artifact.import_data("SampleData[Contigs]", _input)
+
+        enhanced_tabular_results_path = os.path.join(
+            self.get_data_path("quast-results"), "enhanced_tabular_results.tsv"
+        )
+
+        with tempfile.TemporaryDirectory() as tmp:
+            test_enhance_path = os.path.join(tmp, "vis_files", "quast_data")
+            exp_vis = MagicMock()
+            # mocking the _visualize_quast from the ctx.get_action
+            mock_action = MagicMock(side_effect=lambda x, **kwargs: (exp_vis,))
+
+            # mocking the context
+            mock_ctx = MagicMock(
+                get_action=lambda action_type, action_name: mock_action
+            )
```

Comment:
You can move all this code to the `MockContext` class you introduced on top of this file (and never used?)